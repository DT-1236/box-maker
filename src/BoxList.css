.BoxList {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.BoxList > * {
  margin: 4px;
}
