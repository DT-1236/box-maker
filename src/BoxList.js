import React, { Component } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import './BoxList.css';
import uuid from 'uuid/v4';

class BoxList extends Component {
  constructor(props) {
    super(props);
    this.state = { boxes: {} };
    this.makeNewBox = this.makeNewBox.bind(this);
    this.removeBox = this.removeBox.bind(this);
  }

  makeNewBox({ backgroundColor, height, width }) {
    const currentBoxList = { ...this.state.boxes };
    this.setState({
      boxes: {
        ...currentBoxList,
        [uuid()]: {
          backgroundColor,
          height: height + 'px',
          width: width + 'px'
        }
      }
    });
  }

  removeBox(uuid) {
    const boxList = { ...this.state.boxes };
    delete boxList[uuid];
    this.setState({ boxes: boxList });
  }

  renderBoxes() {
    const boxes = [];
    for (let uuid in this.state.boxes) {
      boxes.push(
        <Box
          key={uuid}
          id={uuid}
          style={this.state.boxes[uuid]}
          remove={this.removeBox}
        />
      );
    }
    return boxes;
  }

  render() {
    return (
      <div className="BoxList">
        <NewBoxForm submit={this.makeNewBox} />
        {this.renderBoxes()}
      </div>
    );
  }
}

export default BoxList;
