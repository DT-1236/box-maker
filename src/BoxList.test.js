import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import BoxList from './BoxList';

describe('smoke tests for mount/shallow', () => {
  it('renders without crashing', () => {
    shallow(<BoxList />);
  });

  it('mounts without crashing', () => {
    mount(<BoxList />);
  });
});

describe('snapshot testing', () => {
  it('passes simple snapshots', () => {
    expect(toJson(shallow(<BoxList />))).toMatchSnapshot();
    expect(toJson(mount(<BoxList />))).toMatchSnapshot();
  });
});

describe('it creates and deletes boxes', () => {
  var wrapper;
  beforeEach(() => {
    //   Make a test box
    wrapper = mount(<BoxList />);
    const heightInput = wrapper.find('[name="height"]');
    const widthInput = wrapper.find('[name="width"]');
    const colorInput = wrapper.find('[name="backgroundColor"]');
    heightInput.simulate('change', { target: { name: 'height', value: 30 } });
    widthInput.simulate('change', { target: { name: 'width', value: 65 } });
    colorInput.simulate('change', {
      target: { name: 'backgroundColor', value: '#678901' }
    });
    const newBoxForm = wrapper.find('form');
    newBoxForm.simulate('submit');
  });

  afterEach(() => {
    //   Remove all boxes
    wrapper.setState({ boxes: {} });
  });

  it('makes boxes', () => {
    expect(Object.keys(wrapper.state().boxes)).toHaveLength(1);
    expect(Object.values(wrapper.state().boxes)[0]).toHaveProperty(
      'backgroundColor',
      '#678901'
    );
    expect(Object.values(wrapper.state().boxes)[0]).toHaveProperty(
      'height',
      '30px'
    );
    expect(Object.values(wrapper.state().boxes)[0]).toHaveProperty(
      'width',
      '65px'
    );

    const box = wrapper.find('.Box');
    const boxStyle = box.props().style;
    expect(boxStyle).toHaveProperty('height', '30px');
    expect(boxStyle).toHaveProperty('width', '65px');
    expect(boxStyle).toHaveProperty('backgroundColor', '#678901');
  });

  it('deletes a box', () => {
    expect(Object.keys(wrapper.state().boxes)).toHaveLength(1);
    const deleteButton = wrapper.find('.delete-button');
    deleteButton.simulate('click');
    expect(Object.keys(wrapper.state().boxes)).toHaveLength(0);
  });
});
