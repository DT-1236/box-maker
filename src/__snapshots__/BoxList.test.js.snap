// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing passes simple snapshots 1`] = `
<div
  className="BoxList"
>
  <NewFormBox
    submit={[Function]}
  />
</div>
`;

exports[`snapshot testing passes simple snapshots 2`] = `
<BoxList>
  <div
    className="BoxList"
  >
    <NewFormBox
      submit={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            htmlFor="backgroundColor"
          >
            Color
          </label>
          <input
            name="backgroundColor"
            onChange={[Function]}
            type="text"
            value="#000000"
          />
          <div
            className="sample"
            style={
              Object {
                "backgroundColor": "#000000",
              }
            }
          />
        </div>
        <div>
          <label
            htmlFor="width"
          >
            Width (px)
          </label>
          <input
            name="width"
            onChange={[Function]}
            type="text"
            value={10}
          />
        </div>
        <div>
          <label
            htmlFor="height"
          >
            Height (px)
          </label>
          <input
            name="height"
            onChange={[Function]}
            type="text"
            value={10}
          />
        </div>
        <button
          name="submit-button"
          type="submit"
        >
          Make a box!
        </button>
      </form>
    </NewFormBox>
  </div>
</BoxList>
`;
